nussl/separation/duet.py

```diff
@@ -113,102 +164,78 @@ def run(self):
+        wmat = np.array(np.tile(np.mat(frequency_vector[1:]).T, (1, self.num_time_bins))) * (2 * np.pi / sample_rate)
+
+        return stft_ch0, stft_ch1, wmat
 
+    def _compute_atn_delay(self):
         # Calculate the symmetric attenuation (alpha) and delay (delta) for each
         # time-freq. point
-        R21 = (stft_ch1 + nussl.constants.EPSILON) / (stft_ch0 + nussl.constants.EPSILON)
-        relative_attenuation = np.abs(R21)  # relative attenuation between the two channels
-        symmetric_attenuation = relative_attenuation - 1 / relative_attenuation
-        relative_delay = -np.imag(np.log(R21)) / (2 * np.pi * frequency_matrix)  # relative delay
+        inter_channel_ratio = (self.stft_ch1 + nussl.constants.EPSILON) / (self.stft_ch0 + nussl.constants.EPSILON)
+        atn = np.abs(inter_channel_ratio)  # relative attenuation between the two channels
+        symmetric_atn = atn - 1 / atn  # symmetric attenuation
+        delay = -np.imag(np.log(inter_channel_ratio)) / (2 * np.pi * self.frequency_matrix)  # relative delay
+        return symmetric_atn, delay
+
+    def make_histogram(self, p, q):
+        """Receives the stft of the two channel mixtures and the frequency matrix to a create
+        a smooth and normalized histogram.
 
-        # What is going on here???
-        # ------------------------------------
+        Parameters:
+        stft_ch0 (complex np.array): a 2D Numpy matrix containing the stft of channel 0
+        stft_ch1 (complex np.array): a 2D Numpy matrix containing the stft of channel 1
+        symmetric_atn (np.array): the symmetric attenuation between two channels
+        delay (np.array): the time delay between 2 channels
+        wmat(np.array): a 2D Numpy matrix containing the frequency matrix of the signal
+
+        Returns:
+            hist (np.array): a smooth and normalized histogram
+            atn_bins (np.array): attenuation bins
+            delay_bins (np.array): delay bins
+        """
         # calculate the weighted histogram
-        p = 1
-        q = 0
-        tfw = (np.abs(stft_ch0) * np.abs(stft_ch1)) ** p * (np.abs(frequency_matrix)) ** q  # time-freq weights
+        time_frequency_weights = (np.abs(self.stft_ch0) * np.abs(self.stft_ch1)) ** p \
+                                 * (np.abs(self.frequency_matrix)) ** q
 
         # only consider time-freq. points yielding estimates in bounds
-        a_premask = np.logical_and(self.attenuation_min < symmetric_attenuation, symmetric_attenuation < self.attenuation_max)
-        d_premask = np.logical_and(self.delay_min < relative_delay, relative_delay < self.delay_max)
+        a_premask = np.logical_and(self.attenuation_min < self.symmetric_atn, self.symmetric_atn < self.attenuation_max)
+        d_premask = np.logical_and(self.delay_min < self.delay, self.delay < self.delay_max)
         ad_premask = np.logical_and(a_premask, d_premask)
 
```

Comment:
I'm not a big fan of a lot of these variable names here. `ad_nzind`, *`_vec`, `tfw_vec`, these are all unclear to me.